pulp_container/app/registry_api.py

```diff
@@ -1214,11 +1224,16 @@ def put(self, request, path, pk=None):
                 )
 
             config_digest = config_layer.get("digest")
-            found_config_blobs = models.Blob.objects.filter(
-                digest=config_digest, pk__in=content_pks
-            )
+            found_config_blobs = models.ConfigBlob.objects.filter(digest=config_digest)
+            # create the artifactless ConfigBlob from Blob if it is not stored yet
             if not found_config_blobs.exists():
-                raise BlobInvalid(digest=config_digest)
+                blob = models.Blob.objects.filter(digest=config_digest)
```

Comment:
What if there is not such a blob? I suppose we can raise `BlobInvalid` in this case. I would put this code in a `transaction` so that we will succeed only if we could initialize a config blob and remove the "placeholder" blob.